src/ucp/rma/basic_rma.c

```diff
@@ -117,29 +185,65 @@ ucp_progress_put_inner(ucp_request_t *req, uct_rkey_t uct_rkey,
                                       req->send.rma.remote_addr,
                                       uct_rkey);
         status = (packed_len > 0) ? UCS_OK : (ucs_status_t)packed_len;
+    } else {
+        uct_iov_t iov;
+
+        /* TODO: leave last fragment for bcopy */
+        packed_len = ucs_min(req->send.length, rma_config->max_put_zcopy);
+        iov.buffer = (void *)req->send.buffer;
+        iov.length = packed_len;
+        iov.count  = 1;
+        iov.memh   = req->send.state.dt.contig.memh;
+        ++req->send.uct_comp.count;
+
+        status = UCS_PROFILE_CALL(uct_ep_put_zcopy,
+                                  ep->uct_eps[req->send.lane], 
+                                  &iov, 1, 
+                                  req->send.rma.remote_addr,
+                                  uct_rkey,
+                                  &req->send.uct_comp);
+        if (status <= 0) {
+            --req->send.uct_comp.count;
+        }
     }
 
     return ucp_rma_request_advance(req, packed_len, status);    
 }
 
 static UCS_F_ALWAYS_INLINE ucs_status_t 
 ucp_progress_get_inner(ucp_request_t *req, uct_rkey_t uct_rkey, 
-                       const ucp_ep_rma_config_t *rma_config)
+                       const ucp_ep_rma_config_t *rma_config, int zcopy)
 {
     ucp_ep_t *ep = req->send.ep;
     ucs_status_t status;
     size_t frag_length;
 
     ++req->send.uct_comp.count;
-    frag_length = ucs_min(rma_config->max_get_bcopy, req->send.length);
-    status = UCS_PROFILE_CALL(uct_ep_get_bcopy,
-                              ep->uct_eps[req->send.lane],
-                              (uct_unpack_callback_t)memcpy,
-                              (void*)req->send.buffer,
-                              frag_length,
-                              req->send.rma.remote_addr,
-                              uct_rkey,
-                              &req->send.uct_comp);
+    if (zcopy) {
```

Comment:
maybe better to have !zcopy as first case - for performance, and consistency with put